server/cpho/views/indicators.py

```diff
@@ -43,12 +43,34 @@ class IndicatorForm(ModelForm):
     class Meta:
         model = Indicator
         fields = "__all__"
-        widgets = {
-            "relevant_period_types": forms.CheckboxSelectMultiple,
-        }
+        # widgets = {
+        #     "relevant_period_types": forms.CheckboxSelectMultiple,
+        # }
+
+    def __init__(self, *args, **kwargs):
+        user = None
+        if "user" in kwargs:
+            user = kwargs.pop("user", None)
+        super(IndicatorForm, self).__init__(*args, **kwargs)
+
+        non_hso_readonly_fields = [
+            "name",
+            "category",
+            "topic",
+            "detailed_indicator",
+            "sub_indicator_measurement",
+            "relevant_dimensions",
+            "relevant_period_types",
+        ]
+        if user and not test_rule("is_admin_or_hso", user):
+            for field in non_hso_readonly_fields:
+                # self.fields[field].disabled = True
+                # self.fields[field].widget.attrs["readonly"] = True
+                self.fields[field].widget.attrs["disabled"] = True
```

Comment:
use [field.disabled](https://docs.djangoproject.com/en/5.0/ref/forms/fields/#disabled) instead. Setting this on the widget will only set the attr on the frontend, but `field.disabled` will do this _and_ ignore incoming input values making it impossible to submit data anyway

Also, rm comments